src/Color.cpp

```diff
@@ -43,6 +43,27 @@ Rgb& Rgb::operator=( Hsv hsv ) {
     return *this;
 }
 
+Rgb Rgb::operator+( Rgb in ){
+    uint16_t red = this->r + in.r;
+    uint8_t red_ = (uint8_t)(red<255) ? red : 255;
+    uint16_t green = this->g + in.g;
+    uint8_t green_ = (uint8_t)(green<255) ? green : 255;
+    uint16_t blue = this->b + in.b;
+    uint8_t blue_ = (uint8_t)(blue<255) ? blue : 255;
+    Rgb output( red_, green_, blue_ );
+    return output;
```

Comment:
It is much better to implement only operator+= and implement operator+ via += - in this way you do not duplicate the code logic.
```.cpp
Rgb Rgb::operator+( Rgb other ) const {
    auto copy = *this;
    copy += other;
    return copy;
}
```